tap_xactly/sync.py

```diff
@@ -1,53 +1,105 @@
+import time
 import singer
-from singer import Transformer, metadata
+from singer import Transformer
 
 # The client name needs to be filled in here
-from tap_xactly.client import XactlyClient
-from tap_xactly.streams import STREAMS
+import tap_xactly.helpers as helpers
 
 LOGGER = singer.get_logger()
 
 
 def sync(config, state, catalog):
-    # Any client required PARAMETERS to hit the endpoint
-    client = XactlyClient(config)
+    total_records = []
+    stream_rps = []
 
     with Transformer() as transformer:
         for stream in catalog.get_selected_streams(state):
-            tap_stream_id = stream.tap_stream_id
-            stream_obj = STREAMS[tap_stream_id](client, state)
-            # replication_key = stream_obj.replication_key
-            stream_schema = stream.schema.to_dict()
-            stream_metadata = metadata.to_map(stream.metadata)
+            # Metrics variables
+            stream_start = time.perf_counter()
+            record_count = 0
 
-            LOGGER.info("Staring sync for stream: %s", tap_stream_id)
+            # tap_stream_id = stream.tap_stream_id
+            stream_obj = helpers.get_stream_object(stream, state, config)
+            # stream_schema = helpers.get_stream_schema(stream)
+            # stream_metadata = helpers.get_stream_metadata(stream)
```

Comment:
As per my comment in `helpers.py`, we could pass `stream` into the constructor of `stream_obj`. This would allow us to aggregate all stream data in one object and not need to remember if the data is in `stream` or `stream_obj` while also cutting down on boilerplate.

Reply:
I also take blame here. We've been using the same cookiecutter for awhile now, and I never really thought about refactoring this section until this PR.